superset-frontend/src/SqlLab/components/ResultSet.tsx

```diff
@@ -166,14 +210,148 @@ export default class ResultSet extends React.PureComponent<
     }
   }
 
+  handleOverwriteDatasetOption(data, option) {
+    this.setState({ datasetToOverwrite: option })
+  }
+
+  handleOverwriteDataset() {
+    const { sql, results } = this.props.query;
+    const { datasetToOverwrite } = this.state;
+
+    SupersetClient.put({
+      endpoint: `api/v1/dataset/${datasetToOverwrite.dataSetId}?override_column=true`,
+      headers: { 'Content-Type': 'application/json' },
+      body: JSON.stringify({
+        sql,
+        columns: results.selected_columns.map(d => ({column_name: d.name}))
+      }),
+    }).then(d => {
+        exploreChart({
+          datasource: `${datasetToOverwrite.dataSetId}__table`,
+          metrics: [],
+          groupby: [],
+          time_range: 'No filter',
+          viz_type: 'table',
+          all_columns: results.selected_columns.map(d => d.name),
+          row_limit: 1000,
```

Comment:
not now, but maybe in the future, it seems like most of these params could be defaults in the original `exploreChart` method.

Reply:
since a lot of these values are used three times in this file, we could alternatively set up some defaults here, too.